### PR TITLE
Exporter: export `classes_` attribute for subestimator of RF

### LINKED ISSUE #663: treelite.sklearn.export_model should preserve sklearn-compatible child estimator class metadata for random forest classifiers
## Summary

`treelite.sklearn.export_model` should export scikit-learn-compatible classifier metadata for child estimators in exported random forest classifiers.

Today, cuML needs a local workaround when converting a `RandomForestClassifier` to scikit-learn: after calling the CPU export path, it manually patches each exported child estimator so that:

- the forest-level `classes_` retains the original class labels
- each child estimator's `classes_` uses sklearn-style encoded class indices (`[0, 1, ..., n_classes - 1]`)
- each child estimator's `n_classes_` matches the parent classifier

This behavior is currently implemented downstream in RAPIDS cuML, but it appears to be a responsibility that belongs in Treelite's sklearn export layer.

## Rationale

scikit-learn stores classifier metadata differently at the forest level vs. the individual child trees. If `treelite.sklearn.export_model` exports the same `classes_` metadata everywhere, downstream projects need ad hoc fixups to restore sklearn-compatible child estimator state.

Moving this logic upstream would:

- keep sklearn-export behavior consistent at the source
- remove downstream patching/hacks in RAPIDS cuML
- improve compatibility for any Treelite consumer exporting sklearn random forest classifiers

## Affected area

- `treelite.sklearn.export_model`
- sklearn random forest classifier export path
- exported child estimator metadata (`classes_`, `n_classes_`)

## Observed downstream workaround

RAPIDS cuML PR #8291 adds a post-processing step in `python/cuml/cuml/ensemble/randomforestclassifier.py` to patch exported child estimators after CPU export.

PR: https://github.com/rapidsai/cuml/pull/8291  
Review comment: https://github.com/rapidsai/cuml/pull/8291#discussion_r3470275711

Requested by: @csadorf

## Expected behavior / acceptance criteria

- Exported sklearn random forest classifiers preserve the original labels at the forest level (`forest.classes_`)
- Exported child estimators use sklearn-compatible encoded labels for `estimator.classes_`
- Exported child estimators expose the correct `estimator.n_classes_`
- Behavior matches a natively fitted scikit-learn `RandomForestClassifier`, including cases with non-contiguous class labels
- Downstream projects such as cuML can remove local metadata patching once they adopt the updated Treelite behavior

### PR DESCRIPTION
**Pull request by chyunsu3**
Closes #663 by adopting the fix from https://github.com/NVIDIA/cuml/pull/8291

### CODE REVIEW

**Comment by chyunsu3** [comment]
@csadorf Can I get a review?

File: python/treelite/sklearn/exporter.py [hunk 1]
```diff
@@ -159,9 +159,19 @@ def _export_tree(
     }
     if subestimator_class is DecisionTreeClassifier:
         if n_targets == 1:
-            subestimator_state["n_classes_"] = n_classes[0]
+            subestimator_state.update(
+                {
+                    "n_classes_": n_classes[0],
+                    "classes_": np.arange(n_classes[0]),
```
Comment (csadorf) [inline python/treelite/sklearn/exporter.py]: I think `classes_` would need to be of dtype `np.float64`, no?


```suggestion
                    "classes_": np.arange(n_classes[0], dtype=np.float64),
```

File: python/treelite/sklearn/exporter.py [hunk 2]
```diff
@@ -159,9 +159,19 @@ def _export_tree(
...
+                }
+            )
         else:
-            subestimator_state["n_classes_"] = n_classes.tolist()
+            subestimator_state.update(
+                {
+                    "n_classes_": n_classes.tolist(),
+                    "classes_": [np.arange(n_classes[i]) for i in range(n_targets)],
```
Comment (csadorf) [inline python/treelite/sklearn/exporter.py]: Same as above.

File: python/treelite/sklearn/exporter.py (hunk 1 above)
Comment (chyunsu3) [reply python/treelite/sklearn/exporter.py]: `classes_` now has dtype `np.float64`.

**Review by csadorf — APPROVED** [review]
